### ISSUE TITLE
extra newlines after front matter introduce off-by-one error in line numbers with `--ignore-front-matter`

### ISSUE THREAD

**Issue by verhovsky** [markdownlint/markdownlint#496]
## Description

Both of these files report that the "MD031 Fenced code blocks should be surrounded by blank lines" is on line 4 but it's actually on line 5 in the second file

````md
---
---
sometext
```
```
````

````md
---
---

sometext
```
```
````

```sh
~/code/learnxinyminutes/source/docs$ cd /tmp
/tmp$ mkdir mdl-test
/tmp$ cd mdl-test/
/tmp/mdl-test$ echo $'---\n---\nsometext\n```\n```\n' > a.md
/tmp/mdl-test$ echo $'---\n---\n\nsometext\n```\n```\n' > b.md
/tmp/mdl-test$ mdl .
./a.md:7: MD012 Multiple consecutive blank lines
./a.md:4: MD031 Fenced code blocks should be surrounded by blank lines
./b.md:8: MD012 Multiple consecutive blank lines
./b.md:5: MD031 Fenced code blocks should be surrounded by blank lines
/tmp/mdl-test$ mdl --ignore-front-matter .
./a.md:7: MD012 Multiple consecutive blank lines
./a.md:4: MD031 Fenced code blocks should be surrounded by blank lines
./b.md:7: MD012 Multiple consecutive blank lines
./b.md:4: MD031 Fenced code blocks should be surrounded by blank lines
```

Expected:
the last two lines should also be
```
./b.md:8: MD012 Multiple consecutive blank lines
./b.md:5: MD031 Fenced code blocks should be surrounded by blank lines
```

For easier copy/pasting:

```bash
cd /tmp
mkdir mdl-test
cd mdl-test/
echo $'---\n---\nsometext\n```\n```\n' > a.md
echo $'---\n---\n\nsometext\n```\n```\n' > b.md
mdl .
```

### Environment
macOS

**MDL Version**
0.13.0
